pkg/runner/test_runner.go

```diff
@@ -81,6 +81,8 @@ type routeResult struct {
 
 // stepResult represents the result of a TestStep, possibly carrying error information
 type stepResult struct {
+	jobID  types.JobID
+	runID  types.RunID
```

Comment:
job ID and run ID have been added here since I need them to emit a fully populated event where part of the information is populated in RunTestStep (the bundle and error) and part of the information are visible in the caller, WaitPipelineCompletion